qiskit_ibm_runtime/session.py

```diff
@@ -301,15 +302,36 @@ def from_id(
             session_id: the id of the session to be created. This must be an already
                 existing session id.
             service: instance of the ``QiskitRuntimeService`` class.
-            backend: instance of :class:`qiskit_ibm_runtime.IBMBackend` class or
-                string name of backend.
+            backend (DEPRECATED): instance of :class:`qiskit_ibm_runtime.IBMBackend` class or
+                string name of backend. Either ``service`` or ``backend`` must be given.
+
+         Raises:
+            IBMInputValueError: If given `session_id` does not exist. or the backend passed in does
+                not match the original session backend.
 
         Returns:
             A new Session with the given ``session_id``
 
         """
         if backend:
             deprecate_arguments("backend", "0.15.0", "Sessions do not support multiple backends.")
+            if isinstance(backend, IBMBackend):
+                backend = backend.name
+
+        if service:
+            response = service._api_client.session_details(session_id)
+            if response:
+                session_backend = response.get("backend_name")
+                if backend and backend != session_backend:
+                    raise IBMInputValueError(
+                        f"The session_id {session_id} was created with backend {session_backend}, "
+                        f"but backend {backend} was given."
+                    )
+                backend = session_backend
+            else:
+                raise IBMInputValueError(f"The session_id {session_id} does not exist.")
+        if not service and not backend:
+            raise IBMInputValueError("Either service or backend must be given.")
```

Comment:
Since `backend` is deprecated, we should just make `service` a required keyword.

Reply:
I agree that `service` should be a required argument, but is there some sort of deprecation policy when it comes to changing parameters from optional to required? Or can we just put in the error message that `service` is now required?

Reply:
But this statement `if not service and not backend:` is also a breaking change since the old code didn't say anything about service being required. 

If service is not given, we should give a deprecation warning like you have, and just initialize one (i.e. `QiskitRuntimeService()`) since that's what `Session()` does if you don't pass a service. Then we can use that to query session details. This now should also check for `mode`.

Reply:
If service is not given and we initialize a new service, `QiskitRuntimeService()`, it appears that it does not have access to previous sessions. Querying session details results in a session not found error.